lib/settings.js

```diff
@@ -481,14 +314,19 @@ class Settings extends EventEmitter {
    *
    * @param {string} keyPath
    * @param {Function} handler
-   * @returns {Observer}
+   * @returns {number}
    * @public
    */
   watch(keyPath, handler) {
     assert.strictEqual(typeof keyPath, 'string', 'First parameter must be a string');
     assert.strictEqual(typeof handler, 'function', 'Second parameter must be a function');
+    const observer = new Observer(this, keyPath, handler, _.get(this._cacheObj, keyPath));
+    this._watchHandlers.push(observer);
+    return observer;
+  }
 
-    return this._watchValueAtKeyPath(keyPath, handler);
+  unwatch(keyPath, handler) {
+    this._watchHandlers = _.reject(f => f.keyPath === keyPath && f.handler === handler);
   }
```

Comment:
Since we're returning observer, should we also have a function unwatch(oserver)?